api/controller/readcsv_controller.go

```diff
@@ -0,0 +1,57 @@
+package controller
+
+import (
+	model "api/models"
+	"encoding/csv"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/unrolled/render"
+)
+
+func ReadCSV(w http.ResponseWriter, r *http.Request) {
+
+	var resp = render.New()
+	var Contestants []model.Contestant
+
+	csvf, err := os.Open("../api/lmd.csv")
```

Comment:
This whole logic to read files would be better placed on its own package, maybe a dedicated package to read from the csv and getting all of its info converted to the model you need, like having a function `func GetContestants() ([]model.Contestant, error)`, then you could use this function here on your handler.

Then, if you have an endpoint to get a contestant by its ID, after receiving your list of contestants from the csv service, you could filter them out here on your handler as well and return the one requested, if none was found then you could return a 404 error indicating that you could find the requestd user